sscanss/editor/designer.py

```diff
@@ -955,3 +959,256 @@ def value(self):
+            for label, line_edits in widgets.items():
+                label.setText('')
+                for line_edit in line_edits:
+                    line_edit.setStyleSheet('')
+            for label, boxes in comboboxes.items():
+                label.setText('')
+                for combobox in boxes:
+                    combobox.setStyleSheet('')
+            return True
+        return False
+
+    def setNewCollimator(self):
+        """ When the 'Add New...' option is chosen in the collimator combobox, clear the text."""
+        self.collimator_name.clear()
+        self.detector_combobox.setCurrentIndex(0)
+        self.x_aperture.clear()
+        self.y_aperture.clear()
+        self.visuals.reset()
+
+    def updateValue(self, json_data, folder_path):
+        """Updates the json data of the component
+
+        :param json_data: instrument json
+        :type json_data: Dict[str, Any]
+        :param folder_path: path to instrument file folder
+        :type folder_path: str
+        """
+        self.reset()
+        self.json = json_data
+        instrument_data = json_data.get('instrument', {})
+        self.collimator_list = instrument_data.get('collimators', [])
+
+        try:
+            collimator_data = self.collimator_list[max(self.collimator_combobox.currentIndex(), 0)]
+        except IndexError:
+            collimator_data = {}
+
+        # Collimators combobox
+        collimators = []
+        for index, data in enumerate(self.collimator_list):
+            name = data.get('name', '')
+            if name:
+                collimators.append(f"Collimator {index + 1}")
+
+        # Rewrite the combobox to contain the new list of collimators, and reset the index to the current value
+        index = max(self.collimator_combobox.currentIndex(), 0)
+        self.collimator_combobox.clear()
+        self.collimator_combobox.addItems([*collimators, self.new_collimator_text])
+        self.collimator_combobox.setCurrentIndex(index)
+        if self.collimator_combobox.currentText() == self.new_collimator_text:
+            self.setNewCollimator()
+
+        # Name field
+        name = collimator_data.get('name')
+        if name is not None:
+            self.collimator_name.setText(name)
+
+        # Detectors combobox
+        detectors = []
+        detectors_data = instrument_data.get('detectors', [])
+        for data in detectors_data:
+            detector_name = data.get('name', '')
+            if detector_name:
+                detectors.append(detector_name)
+        self.detector_combobox.clear()
+        self.detector_combobox.addItems([*detectors])
+
+        # Choose a detector if at least one is listed, otherwise display a warning
+        if detectors:
+            detector = collimator_data.get('detector')
+            if isinstance(detector, str):
+                self.detector_combobox.setCurrentText(detector)
+            else:
+                self.detector_combobox.setCurrentIndex(0)
```

Comment:
If the user types in a new detector, the first detector is always selected which is confusing. It might be better to make the combobox editable so the user can type in any detector name (although they would have to go back and create the detector). Also with the editable combobox you don't need the **no detector** warning but will still have to check that the detector value is not empty. What do you think?

Reply:
Yeah, this all sounds good. The big hole is of course that a detector can be specified without existing in the file, but there is a warning given about that, and this way of doing things has the advantage that the detector and collimators can be added in either order.

Reply:
One thing to consider is in a new detector is added via a collimator, but is not defined within the detectors. In this case should the new detector be available as an option within the combobox in the collimator component, or should the combobox be restricted to detectors defined in the detector part of the JSON?